### PR TITLE
[WIP] Bitwise OR for RAJA reductions

### PR DESCRIPTION
**Pull request by Parqua**
# Summary

This PR is a feature addition. It adds Bitwise AND and OR as valid operators for RAJA reductions (only in sequential execution, to be expanded upon review) based completely upon the existing reduction operations. It also adds an identity method to the bitwise AND and OR operators in RAJA, to fulfill the requirements of the reduction patterns. Also included are tests, based upon the existing reduce-basic suite, to test all the backends.

### CODE REVIEW

**Comment by rhornung67** [comment]
Folks,  Kae's summer internship ends ends this week. Can we get some folks to review this PR?

File: include/RAJA/util/Operators.hpp [hunk 1]
```diff
@@ -425,8 +427,11 @@ struct bit_and : public detail::binary_function<Arg1, Arg2, Ret> {
   {
     return lhs & rhs;
   }
+
+RAJA_HOST_DEVICE static constexpr Ret identity() { return Ret{~0}; }
```
Comment (willkill07) [inline include/RAJA/util/Operators.hpp:431]: I think this needs to be `~Ret{0}`. It works fine with signed type casting, but not unsigned

See this godbolt: https://godbolt.org/z/rnE9d8

**Comment by rhornung67** [comment]
@Parqua Reduction tests in the develop branch have been restructured so there are merge conflicts with the PR that will need to be addressed. Will you have time to update your branch to develop? If not, just address review comments related to your changes and I'll work on getting this merged.  Thanks.

File: include/RAJA/util/Operators.hpp (hunk 1 above)
Comment (Parqua) [reply include/RAJA/util/Operators.hpp:431]: Oh, thank you, that is exactly what I needed. I'll also remember this tool.

**Comment by Parqua** [comment]
> 
> 
> @Parqua Reduction tests in the develop branch have been restructured so there are merge conflicts with the PR that will need to be addressed. Will you have time to update your branch to develop? If not, just address review comments related to your changes and I'll work on getting this merged. Thanks.

@rhornung67  Can you point me to the PR implementing the new test format? I don't have any merge conflicts when updating my own branch to develop, so I'm not sure what I need to change at-a-glance.

**Comment by rhornung67** [comment]
@Parqua it looks like your good on the tests. I was confused based on the last sentence in your PR summary above.

File: include/RAJA/util/Operators.hpp [hunk 2]
```diff
@@ -428,7 +428,7 @@ struct bit_and : public detail::binary_function<Arg1, Arg2, Ret> {
     return lhs & rhs;
   }
 
-RAJA_HOST_DEVICE static constexpr Ret identity() { return Ret{~0}; }
+RAJA_HOST_DEVICE static constexpr Ret identity() { return ~Ret{0}; }
```
Comment (rhornung67) [inline include/RAJA/util/Operators.hpp:431]: Yup. That makes more sense!

File: include/RAJA/util/Operators.hpp (hunk 1 above)
Comment (Parqua) [reply include/RAJA/util/Operators.hpp]: Implemented here: https://github.com/Parqua/RAJA/commit/acafb8aa149e8df070b9047afa71bfe2663e2328

**Pull request merged**